Source/AwsCommonRuntimeKit/mqtt/Mqtt5.swift

```diff
@@ -0,0 +1,1053 @@
+
+    /// The subscription identifiers of all the subscriptions this message matched.
+    var subscriptionIdentifiers: [UInt32]? // ignore attempts to set but provide in received packets
+
+    /// Property specifying the content type of the payload.  Not internally meaningful to MQTT5.
+    var contentType: String?
+
+    /// Array of MQTT5 user properties included with the packet.
+    var userProperties: [UserProperty]?
+
+    init(qos: QoS, topic: String) {
+        self.qos = qos
+        self.topic = topic
+    }
+
+    /// Get payload converted to a utf8 String
+    func payloadAsString() -> String? {
+        if let data = payload {
+            return String(data: data, encoding: .utf8)
+        }
+        return nil
+    }
+}
+
+/// "Data model of an `MQTT5 PUBACK <https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901121>`_ packet
+public class PubackPacket {
+
+    /// Success indicator or failure reason for the associated PUBLISH packet.
+    let reasonCode: PubackReasonCode
+
+    /// Additional diagnostic information about the result of the PUBLISH attempt.
+    var reasonString: String?
+
+    /// Array of MQTT5 user properties included with the packet.
+    var userProperties: [UserProperty]?
+
+    init (reasonCode: PubackReasonCode) {
+        self.reasonCode = reasonCode
+    }
+}
+
+/// Configures a single subscription within a Subscribe operation
+public class Subscription {
+
+    /// The topic filter to subscribe to
+    let topicFilter: String
+
+    /// The maximum QoS on which the subscriber will accept publish messages
+    let qos: QoS
+
+    /// Whether the server will not send publishes to a client when that client was the one who sent the publish
+    var noLocal: Bool?
+
+    /// Whether messages sent due to this subscription keep the retain flag preserved on the message
+    var retainAsPublished: Bool?
+
+    /// Whether retained messages on matching topics be sent in reaction to this subscription
+    var retainHandlingType: RetainHandlingType?
+
+    init (topicFilter: String, qos: QoS) {
+        self.topicFilter = topicFilter
+        self.qos = qos
+    }
+}
+
+/// Data model of an `MQTT5 SUBSCRIBE <https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901161>`_ packet.
+public class SubscribePacket {
+
+    /// Array of topic filters that the client wishes to listen to
+    var subscriptions: [Subscription]
+
+    /// The positive int to associate with all topic filters in this request.  Publish packets that match a subscription in this request should include this identifier in the resulting message.
+    var subscriptionIdentifier: UInt32?
+
+    /// Array of MQTT5 user properties included with the packet.
+    var userProperties: [UserProperty]?
+
+    init (subscription: Subscription) {
+        self.subscriptions = [subscription]
+    }
+
+    init (subscriptions: [Subscription]) {
```

Comment:
API Discussion: 
I always feel annoying to create a `Subscription` first, then append it to a `SubscribePacket`. I think we can improve that creation process how. Regardless, it might not be this PR related, we could update it once we have the API design settle.

Reply:
That sounds good. Let's add an overload that takes the topic and qos and builds a sub packet using those as well.